debezium-server/debezium-server-redis/src/main/java/io/debezium/server/redis/RedisDatabaseHistory.java

```diff
@@ -0,0 +1,222 @@
+            })
+                    .onFailure().invoke(
+                            f -> {
+                                LOGGER.warn("Writing to database history stream failed with " + f);
+                                LOGGER.warn("Will retry");
+                            })
+                    .onFailure(JedisConnectionException.class).invoke(
+                            f -> {
+                                LOGGER.warn("Attempting to reconnect to redis ");
+                                this.connect();
+                            })
+                    // retry on failure with backoff
+                    .onFailure().retry().withBackOff(Duration.ofMillis(initialRetryDelay), Duration.ofMillis(maxRetryDelay)).indefinitely()
+                    // write success trace message
+                    .invoke(
+                            item -> {
+                                LOGGER.trace("Record written to database history in redis: " + line);
+                            })
+                    .await().indefinitely();
+        }
+        catch (IOException e) {
+            logger.error("Failed to convert record to string: {}", record, e);
+        }
+    }
+
+    @Override
+    public void stop() {
+        running.set(false);
+        super.stop();
+    }
+
+    @Override
+    protected synchronized void recoverRecords(Consumer<HistoryRecord> records) {
+        List<StreamEntry> entries = Uni.createFrom().item(
+                () -> {
+                    return (List<StreamEntry>) client.xrange(
+                            this.redisKeyName, (StreamEntryID) null, (StreamEntryID) null);
+                })
+                .onFailure().invoke(
+                        f -> {
+                            LOGGER.warn("Reading from database history stream failed with " + f);
+                            LOGGER.warn("Will retry");
+                        })
+                .onFailure(JedisConnectionException.class).invoke(
+                        f -> {
+                            LOGGER.warn("Attempting to reconnect to redis ");
+                            this.connect();
+                        })
+                // retry on failure with backoff
+                .onFailure().retry().withBackOff(Duration.ofMillis(initialRetryDelay), Duration.ofMillis(maxRetryDelay)).indefinitely()
+                .await().indefinitely();
+        for (StreamEntry item : entries) {
+            try {
+                records.accept(new HistoryRecord(reader.read(item.getFields().get("schema"))));
+            }
+            catch (IOException e) {
+                logger.error("Failed to convert record to string: {}", item, e);
+            }
+        }
+
+    }
+
+    @Override
+    public boolean storageExists() {
+        return true;
+    }
+
+    @Override
+    public boolean exists() {
+        return true;
```

Comment:
Can you check with the implementation of `KafkaDatabaseHistory`, there we return true only if there's at least one entry, otherwise that history is assumed to be empty, which in turn triggers a (schema) snapshot.

Reply:
I checked and it seems that the Oracle and MySQL tasks check the offset first. I am not sure how it will behave if the topic exists and is empty. Seems it will throw a warning? From what I can see it may be safer to have both return "true". The storageExists is used in Kafka to set up the topic, which is not needed for Redis. The "exists()" method does seem to return false if the topic is empty, which I am not sure is the desired behavior. Can you also take a look in the MySQLTask class and let me know what you think?

Reply:
> The "exists()" method does seem to return false if the topic is empty, which I am not sure is the desired behavior. 

Yes, that is exactly the desired behavior. The idea is to populate the history if it is empty in case it is a) possible (source DB still has the right information in the TX log) and b) a snapshot mode is used which does a schema snapshot in this case.

Reply:
done